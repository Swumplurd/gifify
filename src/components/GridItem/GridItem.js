import PropTypes from 'prop-types';

export const GridItem = ({title, url, date}) => {
    return (
        <div className="col">
            <div className="card h-100">
                <img src={url} className="card-img-top" alt={url}/>
                <div className="card-body">
                    <h5 className="card-title">{title || 'Sin titulo'}</h5>
                    <a href={url} target="_blank" rel="noopener noreferrer" className="btn btn-secondary">Ver imagen</a>
                </div>
                <div className="card-footer">
                    <small className="text-muted">Fecha y hora de subida: {date}</small>
                </div>
            </div>
        </div>
    )
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
}