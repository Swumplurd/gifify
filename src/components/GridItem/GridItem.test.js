import {GridItem} from './GridItem';
import {shallow} from 'enzyme';

describe('Pruebas en GridItem', () => {
    const title = 'Hola';
    const url = 'https://www';
    const date = '2015-12-12'
    const wrapper = shallow(<GridItem title={title} url={url} date={date} />)

    test('Debe mostrar GridItem correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })  

    test('Debe de tener la img con las propiedades src y alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(url)
    })

    test('la etiqueta small debe contener la clase test-mutted', () => {
        const small = wrapper.find('small')
        expect(small.prop('className')).toBe('text-muted')
        expect(small.prop('className').includes('text-muted')).toBe(true)
    })
    
    
})