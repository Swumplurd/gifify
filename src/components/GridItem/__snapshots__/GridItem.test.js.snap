// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GridItem Debe mostrar GridItem correctamente 1`] = `
<div
  className="col"
>
  <div
    className="card h-100"
  >
    <img
      alt="https://www"
      className="card-img-top"
      src="https://www"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Hola
      </h5>
      <a
        className="btn btn-secondary"
        href="https://www"
        rel="noopener noreferrer"
        target="_blank"
      >
        Ver imagen
      </a>
    </div>
    <div
      className="card-footer"
    >
      <small
        className="text-muted"
      >
        Fecha y hora de subida: 
        2015-12-12
      </small>
    </div>
  </div>
</div>
`;
